core/src/test/java/com/sequenceiq/cloudbreak/reactor/CollectDownscaleCandidatesHandlerTest.java

```diff
@@ -0,0 +1,161 @@
+        Event<CollectDownscaleCandidatesRequest> event = new Event<>(request);
+
+        when(stackService.getByIdWithLists(anyLong())).thenReturn(stack);
+        collectDownscaleCandidatesHandler.accept(event);
+        verify(eventBus).notify(eq(EventSelectorUtil.failureSelector(CollectDownscaleCandidatesResult.class)),
+                argThat(new CollectDownscaleCandidatesResultMatcher(Collections.emptySet(), CollectDownscaleCandidatesHandler.ERROR_STATUS_REASON)));
+    }
+
+    @Test
+    public void shouldCollectDownscaleCandidatesWhenFqdnIsMissing() throws CloudbreakException {
+        int scalingAdjustment = 2;
+        CollectDownscaleCandidatesRequest request
+                = new CollectDownscaleCandidatesRequest(STACK_ID, HOST_GROUP_NAME, scalingAdjustment, Collections.singleton(""));
+        Event<CollectDownscaleCandidatesRequest> event = new Event<>(request);
+
+        when(stackService.getByIdWithLists(anyLong())).thenReturn(stack);
+        collectDownscaleCandidatesHandler.accept(event);
+        verify(ambariDecommissioner).collectDownscaleCandidates(stack, HOST_GROUP_NAME, scalingAdjustment);
+    }
+
+    @Test
+    public void shouldVerifyNodeCountWhenNothingIsMissing() throws CloudbreakException {
+        int scalingAdjustment = 2;
+        String fqdn = "fqdn";
+        CollectDownscaleCandidatesRequest request
+                = new CollectDownscaleCandidatesRequest(STACK_ID, HOST_GROUP_NAME, scalingAdjustment, Collections.singleton(fqdn));
+        Event<CollectDownscaleCandidatesRequest> event = new Event<>(request);
+
+        when(stackService.getByIdWithLists(anyLong())).thenReturn(stack);
+        collectDownscaleCandidatesHandler.accept(event);
+        verify(ambariDecommissioner).verifyNodeCount(stack, stack.getCluster(), fqdn);
+    }
+
+    private static class CollectDownscaleCandidatesResultMatcher extends ArgumentMatcher<Event<CollectDownscaleCandidatesResult>> {
+
+        private final Set<String> expectedHostNames;
+
+        private final String expectedStatusReason;
+
+        private final StringBuilder statusDescription = new StringBuilder();
+
+        CollectDownscaleCandidatesResultMatcher(Set<String> expectedHostNames, String expectedStatusReason) {
+            this.expectedHostNames = expectedHostNames;
+            this.expectedStatusReason = expectedStatusReason;
+        }
+
+        @Override
+        public boolean matches(Object argument) {
+            Event event = (Event) argument;
+            CollectDownscaleCandidatesResult payload = (CollectDownscaleCandidatesResult) event.getData();
+            boolean matches = areHostNamesMatching(payload);
+            String statusReason = payload.getStatusReason();
+            if (!StringUtils.equals(expectedStatusReason, statusReason)) {
+                statusDescription
+                        .append(String.format("Expected and actual statusReason does not match. %n expected=%s, %n actual=%s",
```

Comment:
You have free characters in the previous line, please try to use 160 chars as possible.